src/providers/node/mod.rs

```diff
@@ -152,6 +159,25 @@ impl NodeProvider {
         Ok(false)
     }
 
+    pub fn uses_corepack(app: &App) -> Result<bool> {
+        let package_json: PackageJson = app.read_json("package.json").unwrap_or_default();
+
+        if package_json.package_manager.is_some() {
+            if package_json
+                .package_manager
+                .expect("Already non-null validated")
+                .starts_with("npm")
+            {
+                // Fall back to just using the system npm version.
+                return Ok(false);
+            }
```

Comment:
Nit: Can use

```rs
if let Some(package_manager) = package_json.package_manger {
  if package_manager.starts_with("npm") {
    // ...
  }
}
```

to avoid using `expect`